### PR TITLE
refactor: Use key instead of name for search pills

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Refactor**

This PR resolves []

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Use key instead of name for search pills - dimatretyak

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Use key instead of name for search pills - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 876d7b05173643ac4a3dd36ed251ede05c081cee
</p>

File: src/lib/Scenes/Search/types.ts [hunk 1]
```diff
@@ -11,8 +11,20 @@ export interface AlgoliaSearchResult {
 export type PillEntityType = "algolia" | "elastic"
 
 export interface PillType {
-  name: string
+  indexName?: string
```
Comment (dimatretyak) [inline src/lib/Scenes/Search/types.ts:14]: I think that such a name is more suitable and suggests what it is responsible for

File: src/lib/Scenes/Search/Search.tsx [hunk 2]
```diff
@@ -208,7 +205,7 @@ export const Search: React.FC<SearchProps> = (props) => {
       <ArtsyKeyboardAvoidingView>
         <InstantSearch
           searchClient={searchClient}
-          indexName={selectedPill.type === "algolia" ? selectedPill.name : ""}
+          indexName={selectedPill.type === "algolia" ? selectedPill.indexName! : ""}
```
Comment (gkartalis) [inline src/lib/Scenes/Search/Search.tsx:208]: 😍

File: src/lib/Scenes/Search/types.ts (hunk 1 above)
Comment (gkartalis) [reply src/lib/Scenes/Search/types.ts:14]: totally agree!

File: src/lib/Scenes/Search/types.ts [hunk 3]
```diff
@@ -11,8 +11,20 @@ export interface AlgoliaSearchResult {
...
   displayName: string
   disabled?: boolean
   type: PillEntityType
+  key: AlgoliaIndiceKey | string
+}
+
+export enum AlgoliaIndiceKey {
```
Comment (gkartalis) [inline src/lib/Scenes/Search/types.ts:21]: **suggestion:** Let's change that to be AlgoliaIndexKey
```suggestion
export enum AlgoliaIndexKey {
```

**Review by gkartalis — APPROVED** [review]
Great job! Just left a comment for a spelling  correction, other than that everything looks great! Thanks!

**Pull request merged**